deeplake/core/transform/test_transform.py

```diff
@@ -1028,3 +1033,126 @@ def test_transform_info(local_ds_generator):
         assert ds.info["test"] == 123
     ds = local_ds_generator()
     assert ds.info["test"] == 123
+
+
+@parametrize_num_workers
+@all_schedulers
+@all_compressions
+@pytest.mark.parametrize(
+    "ds",
+    ["memory_ds", "local_ds", "s3_ds"],
+    indirect=True,
+)
+def test_read_only_dataset_aggregation_image(
+    ds, sample_compression, scheduler, num_workers
+):
+    if scheduler == "processed" and ds.path.startswith("mem://"):
+        pytest.xfail("Memory dataset not supported in processed scheduler")
+
+    i_start = 0
+    i_stop = 100
+    with ds:
+        ds.create_tensor("image", htype="image", sample_compression=sample_compression)
+        for i in range(i_start, i_stop):
+            ds.image.append(i * np.ones((9, 16), dtype="uint8"))
+    ds.read_only = True
+
+    values = []
+    fn_aggregate(key="image", values=values).eval(
+        ds,
+        num_workers=num_workers,
+        progressbar=False,
+        scheduler=scheduler,
+        read_only_ok=True,
+    )
+    assert len(values) == i_stop - i_start
+    assert np.array(values).mean() == (i_start + i_stop - 1) / 2  # half-open interval
+
+    ds._locked_out = True
+    ds.read_only = False
+    ds.delete()
```

Comment:
we don't need these 3 lines at the end. The dataset fixtures are automatically cleaned up